src/aal/aal_powerpc.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#if defined(__powerpc64__)
+#  define SNMALLOC_VA_BITS_64
+#else
+#  define SNMALLOC_VA_BITS_32
+#endif
+
+namespace snmalloc
+{
+  /**
+   * ARM-specific architecture abstraction layer.
+   */
+  class AAL_powerpc
+  {
+  public:
+    /**
+     * Bitmap of AalFeature flags
+     */
+    static constexpr uint64_t aal_features = IntegerPointers;
+
+    /**
+     * On pipelined processors, notify the core that we are in a spin loop and
+     * that speculative execution past this point may not be a performance gain.
+     */
+    static inline void pause()
+    {
+      __asm__ volatile("or 27,27,27"); // "yield"
```

Comment:
Thank you for reminding me why I hate PowerPC assembly so much.

Reply:
I'm not sure why there isn't a pseudoinstruction hint in LLVM, but so it goes.  This is no worse than any other "idempotent operator's encoding becomes hint instead" that everybody and their brother loves, is it?

Reply:
It's more the complete lack of syntactic hint to let you know that 27 is a register number or a literal that I dislike.  PowerPC assembly seems designed to discourage anyone from ever reading any, let alone writing some.